apps/wallet/src/ui/app/pages/approval-request/transaction-request/TransactionDetails/Command.tsx

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { ChevronDown12, ChevronRight12 } from '@mysten/icons';
+import {
+    type CommandArgument,
+    formatAddress,
+    type TransactionCommand,
+} from '@mysten/sui.js';
+import { useState } from 'react';
+
+import { Text } from '_src/ui/app/shared/text';
+
+function convertCommandArgumentToString(
+    arg: string | string[] | CommandArgument | CommandArgument[]
+): string {
+    if (typeof arg === 'string') return arg;
+
+    if (Array.isArray(arg)) {
+        return `[${arg
+            .map((argVal) => convertCommandArgumentToString(argVal))
+            .join(', ')}]`;
+    }
+
+    switch (arg.kind) {
+        case 'GasCoin':
+            return 'GasCoin';
+        case 'Input':
+            return `Input(${arg.index})`;
+        case 'Result':
+            return `Result(${arg.index})`;
+        case 'NestedResult':
+            return `NestedResult(${arg.index}, ${arg.resultIndex})`;
+        default:
+            // NOTE: This doesn't throw to help keep forwards compatibility:
+            return 'Unknown';
+    }
+}
+
+function convertCommandToString({ kind, ...command }: TransactionCommand) {
+    const commandArguments = Object.entries(command);
+
+    return commandArguments
+        .map(([key, value]) => {
+            if (key === 'target') {
+                const [packageId, moduleName, functionName] = value.split('::');
+                return [
+                    `package: ${formatAddress(packageId)}`,
+                    `module: ${moduleName}`,
+                    `function: ${functionName}`,
+                ].join(', ');
+            }
+
+            return `${key}: ${convertCommandArgumentToString(value)}`;
+        })
+        .join(', ');
+}
+
+export function Command({ command }: { command: TransactionCommand }) {
+    const [expanded, setExpanded] = useState(true);
+
+    return (
+        <div>
+            <div
+                role="button"
```

Comment:
I don't think this is accessible despite `role="button"` because you'd need to create click/keydown handlers to handle focus states properly (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role). Can we just use the `button` element?

Reply:
Oops good catch, meant to do this before opening a PR just was getting lazy with resetting the styling and forgot! Will update